File: src/_03_BehavioralPattern/_03_04_Iterator/java/after/Board.java
package _03_BehavioralPattern._03_04_Iterator.java.after;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

public class Board {
  
  List<Post> posts = new ArrayList<>();

  public List<Post> getPosts() {
    return this.posts;
  }
  public void setPosts(List<Post> posts) {
    this.posts = posts;
  }

  public void addPost(String content) {
    this.posts.add(new Post(content));
  }

  public Iterator<Post> getRecentPostIterator() {
    return new RecentPostIterator(this.posts);
  }
  
  public Iterator<Post> getAscendingPostIterator() {
    return new AscendingPostIterator(this.posts);
  }
  
}

File: src/myTest/Iterator2/Board.js
import Post from "./Post.js";

export default class Board {
  /** @type { Post[] } posts */
  posts;

  constructor() {
    this.posts = [];
  }

  /** @returns { Post[] } posts */
  getPosts() {
    return this.posts;
  }
  /** @param { Post[] } posts */
  setPosts(posts) {
    this.posts = posts;
  }

  /** @param { string } content */
  addPost(content) {
    this.posts.push(new Post(content));
  }
}